pkg/sync/sync_context.go

```diff
@@ -597,6 +599,14 @@ func (sc *syncContext) autoCreateNamespace(tasks syncTasks) syncTasks {
 					tasks = append(tasks, nsTask)
 				} else {
 					sc.log.Infof("Namespace %s is already existed.", sc.namespace)
+					//We need to remove app label to avoid prunning the namespace
+					if liveObj != nil && sc.appLabelKey != "" && kube.GetAppInstanceLabel(liveObj, sc.appLabelKey) != "" {
+						kube.UnsetLabel(liveObj, sc.appLabelKey)
+						_, err := sc.kubectl.ApplyResource(context.TODO(), sc.rawConfig, liveObj, liveObj.GetNamespace(), cmdutil.DryRunNone, false, true)
```

Comment:
We still need to leverage `syncTasks` to make sure apply is executed only if dry run was successful and permission checks passing. 

Instead of manually executing apply we can create sync task where `target` object is live namespace copy with modified labels.

Reply:
👍